rust-runtime/aws-smithy-runtime-api/src/client/http.rs

```diff
@@ -165,6 +166,15 @@ pub trait HttpClient: Send + Sync + fmt::Debug {
         let _ = (runtime_components, cfg);
         Ok(())
     }
+
+    /// Provide metadata about the crate that this HttpClient uses to make connectors.
+    ///
+    /// If this is implemented and returns metadata, that metadata will be
+    /// inserted into the user agent string when making a request with this
+    /// client.
```

Comment:
To verify my understanding, is it still true when an SDK is code-generated without AWS codegen decorators?

Reply:
It isn't b/c I don't think we have a generic user agent interceptor. I'll re-word this.